src/context/like-context.js

```diff
@@ -0,0 +1,46 @@
+import { useReducer, useContext, createContext } from "react";
+
+const likeContext = createContext(null);
+
+const LikeContextProvider = ({ children }) => {
+  const reducerFunction = (likeState, { type, payload }) => {
+    switch (type) {
+      case "LIKE":
+        let index = likeState.likeList.findIndex(
```

Comment:
you can write the reducer function in a separate folder to maintain good and clean code.

Reply:
Okay